ModelBundle/Tests/Functional/Repository/NodeRepositoryTest.php

```diff
@@ -373,72 +373,47 @@ public function testFindLastPublishedVersion($language, $siteId, $count)
     }
 
     /**
-     * @return array
-     */
-    public function provideLanguageSiteIdAndCount()
-    {
-        return array(
-            array('en', '2', 5),
-            array('fr', '2', 5),
-        );
-    }
-
-
-    /**
      * @param string       $author
+     * @param string       $siteId
      * @param boolean|null $published
+     * @param int          $limit
+     * @param array|null   $sort
      * @param int          $count
      *
-     * @dataProvider provideFindByAuthor
+     * @dataProvider provideFindByAuthorAndSiteId
      */
-    public function testFindByAuthor($author, $published, $count)
+    public function testFindByAuthorAndSiteId($author, $siteId, $published, $limit, $sort, $count)
     {
-        $this->assertCount($count, $this->repository->findByAuthor($author, $published));
+        $this->assertCount(
+            $count,
+            $this->repository->findByAuthorAndSiteId($author, $siteId, $published, $limit, $sort)
+        );
     }
 
     /**
      * @return array
      */
-    public function provideFindByAuthor()
+    public function provideLanguageSiteIdAndCount()
     {
         return array(
-            array('fake_admin', null, 5),
-            array('fake_admin', false, 1),
-            array('fake_admin', true, 4),
-            array('fakeContributor', false, 0),
-            array('fakeContributor', null, 0),
+            array('en', '2', 5),
+            array('fr', '2', 5),
         );
     }
 
     /**
-     * @param string       $author
-     * @param string       $siteId
-     * @param boolean|null $published
-     * @param array|null   $sort
-     * @param int          $count
-     *
-     * @dataProvider provideFindByAuthorAndSiteId
-     */
-    public function testFindByAuthorAndSiteId($author, $siteId, $published, $sort, $count)
-    {
-        $this->assertCount(
-            $count,
-            $this->repository->findByAuthorAndSiteId($author, $siteId, $published, $sort)
-            );
-    }
-
-    /**
      * @return array
      */
     public function provideFindByAuthorAndSiteId()
     {
         return array(
-            array('fake_admin', '2', null, array('updatedAt' => -1), 4),
-            array('fake_admin', '2', false, null, 1),
-            array('fake_admin', '2', true, null, 3),
-            array('fakeContributor', '2', false, null, 0),
-            array('fakeContributor', '2', null, null, 0),
-            array('fake_admin', '3', true, null, 1),
+            array('fake_admin', '2', null, 10, array('updatedAt' => -1), 4),
+            array('fake_admin', '2', false, 10, null, 1),
+            array('fake_admin', '2', true, 10, null, 3),
+            array('fake_admin', '2', true, 2, null, 2),
+            array('fakeContributor', '2', false, 10, null, 0),
+            array('fakeContributor', '2', null, 10, null, 0),
+            array('fake_admin', '3', true, 10, null, 1),
         );
```

Comment:
I think you should standardize authors : fake_admin and fake_contributor or fakeAdmin and fakeContributor